src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NonConstantFieldsShouldNotBeVisible.cs

```diff
@@ -32,8 +33,23 @@ public abstract class NonConstantFieldsShouldNotBeVisibleAnalyzer : DiagnosticAn
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
-        { 
-            
+        {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                if (!field.IsPublic())
+                {
+                    return;
+                }
+
+                if (field.IsConst || (field.IsStatic && field.IsReadOnly))
+                {
+                    return;
+                }
+
+                obj.ReportDiagnostic(field.CreateDiagnostic(Rule));
```

Comment:
Ya we could just issue CA1051 from this analyzer. @twsouthwick - is that something you can do? Issue that diagnostic from here and rename the files to be a bit more generic and delete the stubs for the CA1051 rule.